hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -292,6 +309,60 @@ private[sql] class AvroSerializer(
       result
   }
 
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  // Following section is amended to the original (Spark's) implementation
+  // >>> BEGINS
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  private def newUnionConverter(catalystStruct: StructType,
```

Comment:
add UTs to cover this?

Reply:
There're UT that cover this

Reply:
are you referring to AvroSerializer being used throughout the spark tests? but i don't see a specific UT for this union converter logic

Reply:
It's `TestAvroSerDe`